database.rules.json

```diff
@@ -147,6 +147,21 @@
         }
       }
     },
+    "jupyterSolutionAnalysisQueue" : {
+      "responses": {
+      "$taskKey": {
+        ".read": "auth != null",
+        ".write": "!newData.exists()"
```

Comment:
This allows anyone to write into this node. If I get it right, only firebase function should write into this node. Firebase function ignores rules restrictions, so it able to write into it even with `".write": false` rule

Reply:
`".write": "!newData.exists()"`
This rule  defined that user have only access to delete the data. Because after reading the response from this path, a user has to delete the response from here.
We can modify the rules as
`".write": "!newData.exists() && data.child('owner').val() == auth.uid"`
So that only owner have access to delete this.